### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by paul-nameless**
- Move update handlers to separate module
- Move extended implementation of telegram api to tdlib.py
- Rewrote keybinding handlers

### CODE REVIEW

File: tg/update_handlers.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+from tg.msg import MsgProxy
+
+log = logging.getLogger(__name__)
+
+_update_handler_type = Callable[[Controller, Dict[str, Any]], None]
+
+handlers: Dict[str, _update_handler_type] = {}
+
+
+def update_handler(update_type):
```
Comment (lightme16) [inline tg/update_handlers.py:16]: cools like an inconsistency. 
for keybindings you are manually registering shortcut handlers in the dict, but here you are using decorator to register it. Maybe stick to one approach?

File: tg/controllers.py [hunk 2]
```diff
@@ -34,6 +34,8 @@
 # cause blan areas on the msg display screen
 MSGS_LEFT_SCROLL_THRESHOLD = 2
 
+key_bind_handler = Callable[[Any], Any]
```
Comment (lightme16) [inline tg/controllers.py:37]: bad type name, should be `key_bind_handler_type`

File: tg/controllers.py [hunk 3]
```diff
@@ -43,38 +45,166 @@ class Controller:
...
+            "q": lambda _: "QUIT",
+            "l": self.handle_msgs,
+            "j": self.next_chat,
+            "^N": self.next_chat,
+            "k": self.prev_chat,
+            "^P": self.prev_chat,
+            "J": lambda _: self.next_chat(10),
+            "K": lambda _: self.prev_chat(10),
+            "gg": self.first_chat,
+            "bp": self.breakpoint,
```
Comment (lightme16) [inline tg/controllers.py:66]: :bulb: to get rig of that terrible `_: int` just use lambda `lambda _: self.breakpoint(),`

File: tg/controllers.py (hunk 2 above)
Comment (paul-nameless) [reply tg/controllers.py:37]: 👌

File: tg/update_handlers.py (hunk 1 above)
Comment (paul-nameless) [reply tg/update_handlers.py:16]: Truly saying, I am not sure what is the best way to make it readable.

As for me, I prefer seeing keybinding in one place though I don't care about update_hanlders.

What's your propositions? Which way you prefer?

**Pull request merged**